### PR TITLE
Bug 1218832 - Rolling File Logger Crash

### PR DESCRIPTION
**Pull request by sleroux**
The crash was being caused by converting a UInt64 swift pointer into a Int64 value in ObjC through the inout parameter of the NSFileManager extension getAllocatedSize method.

* Rewrote NSFileManagerExtension in Swift. 
* Cleared up method name in RollingFileLogger to make prefix more obvious. 
* Cleaned up test try logic.
* Removed NSFileManager ObjC extension.

### CODE REVIEW

File: SharedTests/RollingFileLoggerTests.swift [hunk 1]
```diff
@@ -45,11 +45,7 @@ class RollingFileLoggerTests: XCTestCase {
     func testNewLogDeletesPreviousLogIfItsTooLarge() {
         let expectedPath = createNewLogFileWithSize(5001)
```
Comment (rnewman) [inline SharedTests/RollingFileLoggerTests.swift]: This really ought to be `self.sizeLimit + 1`.

File: Utils/Extensions/NSFileManagerExtensions.swift [hunk 2]
```diff
@@ -0,0 +1,95 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
```
Comment (rnewman) [inline Utils/Extensions/NSFileManagerExtensions.swift]: Fix leading indent. (*shakes fist at Xcode*)

File: Utils/Extensions/NSFileManagerExtensions.swift [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            guard let isRegular = isRegularFile as? Bool
+                where isRegular == true && (itemURL.pathComponents?.last?.hasPrefix(prefix) ?? false) else {
+                continue
+            }
+
+            // First try to get the total allocated size and in failing that, get the file allocated size
+            var fileSize: AnyObject?
+            try itemURL.getResourceValue(&fileSize, forKey: NSURLTotalFileAllocatedSizeKey)
+            if fileSize == nil {
+                try itemURL.getResourceValue(&fileSize, forKey: NSURLFileAllocatedSizeKey)
```
Comment (rnewman) [inline Utils/Extensions/NSFileManagerExtensions.swift]:
```
extension NSURL {
    public func getResourceValueForKey(key: String) -> AnyObject? {
        var val: AnyObject?
        …
        return val
    }
```

Comment (rnewman) [inline Utils/Extensions/NSFileManagerExtensions.swift]:
```
     private func getResourceLongForKey(key: String) -> Int64? {
```

File: Utils/Extensions/NSFileManagerExtensions.swift [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+            guard let itemURL = itemURL as? NSURL else {
+                throw errorWithCode(.EnumeratorElementNotURL)
+            }
+
+            var isRegularFile: AnyObject?
+            try itemURL.getResourceValue(&isRegularFile, forKey: NSURLIsRegularFileKey)
+
+            // Skip files that are not regular and don't match our prefix
+            guard let isRegular = isRegularFile as? Bool
+                where isRegular == true && (itemURL.pathComponents?.last?.hasPrefix(prefix) ?? false) else {
```
Comment (rnewman) [inline Utils/Extensions/NSFileManagerExtensions.swift]:
```
guard (isRegularFile as? Bool) == true …
```

or add an extension:

```
extension NSURL {
    public var isRegularFile: Bool {
        var isRegularFile: AnyObject?
        try self.getResourceValue(&isRegularFile, …)
        return isRegularFile as? Bool ?? false
    }
```

**Comment by rnewman** [comment]
LGTM, but I'd like to see more of the logic lifted out into extension methods. Each should be obvious, and it'll make the logic itself obvious, too.

File: Utils/Extensions/NSFileManagerExtensions.swift [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+            throw errorWithCode(.ErrorEnumeratingDirectory, underlyingError: enumeratorError)
+        }
+
+        for itemURL in directoryEnumerator {
+            guard let itemURL = itemURL as? NSURL else {
+                throw errorWithCode(.EnumeratorElementNotURL)
+            }
+
+            // Skip files that are not regular and don't match our prefix
+            guard itemURL.isRegularFile && itemURL.lastComponentIsPrefixedBy(prefix) else {
```
Comment (rnewman) [inline Utils/Extensions/NSFileManagerExtensions.swift]: Beautiful.

**Comment by sleroux** [comment]
Closing since work from https://github.com/mozilla/firefox-ios/pull/1237 include this.

**Pull request closed without merging**